lib/src/solver/incompatibility_cause.dart

```diff
@@ -96,6 +97,11 @@ class SdkCause extends IncompatibilityCause {
 
   @override
   String? get hint {
+    if (sdk.isDartSdk &&
+        !LanguageVersion.fromSdkConstraint(constraint).supportsNullSafety &&
+        sdk.version! >= Version(3, 0, 0).firstPreRelease) {
+      return 'The constraint $constraint does not support null-safety.';
```

Comment:
I think this may be easier to understand if we only mention the lower constraint?